tests/test_github.py

```diff
@@ -52,6 +54,51 @@ def mock_input(msg: str) -> str:
     return '1'
 
 
+prs_per_page = 20
+number_of_pages = 3
+
+
+def mock_read(self: Any) -> bytes:
+    response_data = [
+        {
+            'head': {'ref': f'feature_{i}', 'repo': {'full_name': 'testing/checkout_prs',
+                                                     'html_url': 'https://github.com/tester/repo_sandbox.git'}},
+            'user': {'login': 'github_user'},
+            'base': {'ref': 'develop'},
+            'number': f'{i}',
+            'html_url': 'www.github.com',
+            'state': 'open'
+        } for i in range(mock_read.counter * prs_per_page, (mock_read.counter + 1) * prs_per_page)]  # type: ignore[attr-defined]
+
+    mock_read.counter += 1  # type: ignore[attr-defined]
+    return json.dumps(response_data).encode()
+
+
+def mock_info(x: Any) -> Dict[str, Any]:
+    if mock_info.counter < number_of_pages - 1:  # type: ignore[attr-defined]
+        link = f'<https://api.github.com/repositories/1300192/pulls?page={mock_info.counter + 2}>; rel="next"'  # type: ignore[attr-defined]
+    else:
+        link = ''
+    mock_info.counter += 1  # type: ignore[attr-defined]
+    return {"link": link}
+
+
+mock_info.counter = mock_read.counter = 0  # type: ignore[attr-defined]
+
+
+def mock_get_local_branches(self) -> List[LocalBranchShortName]:
+
+    def atoi(text: str) -> Any:
+        return int(text) if text.isdigit() else text
+
+    def natural_keys(text: str) -> List[Any]:
+        return [atoi(c) for c in re.split(r'(\d+)', text)]
+
+    if self._local_branches_cached is None:
+        self._load_branches()
+    return sorted(self._local_branches_cached, key=natural_keys)
+
+
```

Comment:
The tests where not passing when number of prs per page was > 10 because of branch name sorting. e.g
```
feature_1
feature_11
feature_12
feature_2
itd
```
which i changed into natural sorting e.g.
```
feature_1
feature_2
...
feature_10
feature_11
itd
```
I actually think that we can add this sorting to the code base in separate PR since it seems to be useful, @PawelLipski wdyt?

Reply:
Huh... this sorting would apply in what places specifically? `discover` and `github checkout-prs` right? I mean... in real-life/non-test scenarios it's extremely rare that natural sorting would ever yield a different result than plain ASCII sortingm I wouldn't give it a high priority 🤔